src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/helpers/CommonHelper.ps1

```diff
@@ -0,0 +1,280 @@
+        $HelmRepoUrl = Get-ChildItem -Path Env:HELMREPOURL
+        helm repo add $HelmRepoName $HelmRepoUrl --kubeconfig $KubeConfig --kube-context $KubeContext
+    }
+
+    $resources = Get-Module Az.Resources -ListAvailable
+    if ($null -eq $resources) {
+        Write-Error "Missing required module(s): Az.Resources. Please run 'Install-Module Az.Resources -Repository PSGallery' to install Az.Resources."
+        return
+    }
+
+    if (Test-Path Env:HELMREGISTRY) {
+        $RegistryPath = Get-ChildItem -Path Env:HELMREGISTRY
+    }
+    else {
+        $ReleaseTrain = ''
+        if ((Test-Path Env:RELEASETRAIN) -and (Test-Path Env:RELEASETRAIN)) {
+            $ReleaseTrain = Get-ChildItem -Path Env:RELEASETRAIN
+        }
+        else {
+            $ReleaseTrain = 'stable'
+        }
+        $AzLocation = Get-AzLocation | Where-Object { ($_.DisplayName -ieq $Location) -or ($_.Location -ieq $Location) }
+        $Region = $AzLocation.Location
+        if ($null -eq $Region) {
+            Write-Error "Invalid location: $Location"
+            return
+        }
+        else {
+            $Location = $Region
+        }
+        $ChartLocationUrl = "https://${Location}.dp.kubernetesconfiguration.azure.com/azure-arc-k8sagents/GetLatestHelmPackagePath?api-version=2019-11-01-preview&releaseTrain=${ReleaseTrain}"
+
+        $Uri = [System.Uri]::New($ChartLocationUrl)
+        $Env = [Microsoft.Azure.Commands.Common.Authentication.Abstractions.AzureEnvironment]::PublicEnvironments[[Microsoft.Azure.Commands.Common.Authentication.Abstractions.EnvironmentName]::AzureCloud]
+        $PromptBehavior = [Microsoft.Azure.Commands.Common.Authentication.ShowDialog]::Never
+        $Token = [Microsoft.Azure.Commands.Common.Authentication.AzureSession]::Instance.AuthenticationFactory.Authenticate($Account, $Env, $TenantId, $null, $PromptBehavior, $null)
+        $AccessToken = $Token.AccessToken
+
+        $HeaderParameter = @{
+            "Authorization" = "Bearer $AccessToken"
+        }
+
+        $Response = Invoke-WebRequest -Uri $Uri -Headers $HeaderParameter -Method Post -UseBasicParsing
+        if ($Response.StatusCode -eq 200) {
+            $RegistryPath = ($Response.Content | ConvertFrom-Json).repositoryPath
+        }
+        else {
+            Write-Error "Error while fetching helm chart registry path: ${$Response.RawContent}"
+            return
+        }
+    }
+    Set-Item -Path Env:HELM_EXPERIMENTAL_OCI -Value 1
+}
+
+function Configure-ProxySettings {
+    param (
+        [string]$HttpProxy,
+        [string]$HttpsProxy,
+        [string]$NoProxy,
+        [string]$ProxyCert,
+        [bool]$DisableAutoUpgrade,
+        [bool]$ProxyEnableState,
+        [string]$ContainerLogPath,
+        [string]$KubeConfig,
+        [string]$KubeContext,
+        [string]$CustomLocationsOid,
+        [hashtable]$ConfigurationSetting,
+        [hashtable]$ConfigurationProtectedSetting,
+        [hashtable]$PSBoundParameters
```

Comment:
I wonder what's the scope of `$PSBoundParameters` and is it really necessary to pass it in?